### PR TITLE
sdn rebase and vagrant fix

### PR DESCRIPTION
**Pull request by rajatchopra**
Takes care of https://github.com/detiber/openshift-ansible/issues/33
And vagrant up should run fine with node registration.

Fixes https://github.com/openshift/origin/issues/2834

### CODE REVIEW

**Comment by rajatchopra** [comment]
@sdodson Rebased the sdn stuff (sysctl etc)
@jwhonce vagrant stuff works with this PR now, review the vagrant/provision-master.sh block please.

File: vagrant/provision-master.sh [hunk 1]
```diff
@@ -107,3 +107,23 @@ fi
 # Set up the OPENSHIFTCONFIG environment variable for use by osc
 echo 'export OPENSHIFTCONFIG=/vagrant/openshift.local.config/master/admin.kubeconfig' >> /root/.bash_profile
 echo 'export OPENSHIFTCONFIG=/vagrant/openshift.local.config/master/admin.kubeconfig' >> /home/vagrant/.bash_profile
+
+# wait for openshift-master to come alive
+export OPENSHIFTCONFIG=/vagrant/openshift.local.config/master/admin.kubeconfig
+while true; do
+  if ! /usr/bin/openshift cli get nodes; then
+    sleep 1
```
Comment (jwhonce) [inline vagrant/provision-master.sh]: @rajatchopra How about adding an echo -n '.' before the sleep so people will know their waiting and not locked up?

**Comment by rajatchopra** [comment]
@jwhonce Added an echo. Just a '.' will not look good because the 'if' check prints the failing command's output too. Added an explicit comment.

**Comment by rajatchopra** [comment]
Rebase takes care of #2834 too.

**Comment by mrunalp** [comment]
[merge]

**Pull request merged**